practice-use-cases/usecase2-sar_dask_chunks.ipynb

```diff
@@ -0,0 +1,200 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Make a list of SLC paths to read\n",
+    "# Explore the files stored in the directory and create a list of the files (format: *_cint_srd.raw) to read\n",
+    "# ~2 lines of code\n",
+    "\n",
+    "# Create two variables with paths to the latitude and longitude files\n",
+    "# Geo referenced coordinates\n",
+    "# ~2 lines of code\n",
+    "f_lat = [path / 'lat.raw']\n",
+    "f_lon = [path / 'lon.raw']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define 2D Chunks for reading in parallel with dask\n",
+    "# ~1 line of code\n",
+    "\n",
+    "# Load the SLC dataset - you can do this using the sarxarray function from_binary\n",
+    "# The function can be called as:\n",
+    "# from_binary(list_of_slc_paths, shape_of_slcs, dtype, dask_chunk_size)\n",
+    "# dtype for this dataset is np.complex64 \n",
+    "# ~1 line of code\n",
+    "\n",
+    "# Load coordinates - this can also be done using the same from_binary function of sarxarray\n",
+    "# ~2 lines of code"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# DO NOT MODIFY\n",
+    "stack = stack.assign_coords(lat = ((\"azimuth\", \"range\"), lat.squeeze().lat.data), lon = ((\"azimuth\", \"range\"), lon.squeeze().lon.data))\n",
+    "if stack.shape != [2000, 4000, 10]:\n",
+    "    print ('Data not read properly, please check your function')\n",
+    "else:\n",
+    "    print ('Data read successfully')\n",
+    "    stack"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Calculate the mean reflectance map"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# You can calculate the mrm using the following command available from the sarxarray library\n",
+    "# stack.slcstack.mrm() \n",
+    "# Use .compute to persist and calculate the values of the mrm.\n",
+    "# ~2 lines of code\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot the mean reflectance map\n",
+    "from matplotlib import pyplot as plt\n",
+    "fig, ax = plt.subplots()\n",
+    "ax.imshow(mrm)\n",
+    "ax.set_aspect(2)\n",
+    "im = mrm.plot(ax=ax, cmap='gray')\n",
+    "im.set_clim([0, 40000])"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Challenge:\n",
+    "\n",
+    "Rerun the above workflow for different chunk sizes and analyse the performance of the dask cluster. See if you can identify an optimal chunk size for this problem for the fixed worker size.\n",
+    "\n",
+    "Hint: Use %%timeit at the top of a jupyter cell to time the execution time for the cell"
```

Comment:
`%%timeit ` runs a cell multiple times and return the average execution time (with std). When running things with dask things might go wrong, as intermediate results might be cached, thus leading to much shorter timings when calculations are rerun. So maybe using `%%time` here  works better?

Reply:
Fixed